### PR TITLE
libxm: fix period calculation of high octaves in Amiga mode

### PR DESCRIPTION
**Pull request by rasky**
In octaves 6 and 7, there was not enough precision left to accurately
compute the period, because too many bits were discarded during the
scaling down.

To fix this, scale the frequency table by 1024 after switching to
uint32_t, and adjust calculations accordingly.

Co-developed with @bryc.

### CODE REVIEW

**Comment by rasky** [comment]
I'm attaching a couple of tests where the effect of the PR can be heard (they sound of pitch without). I'll leave it up to you whether to commit them or not in this case, since it's a pitch issue.

[pitch_issue.zip](https://github.com/Artefact2/libxm/files/6343119/pitch_issue.zip)

File: src/play.c [hunk 1]
```diff
@@ -304,7 +306,7 @@ static float xm_amiga_period(float note) {
 	unsigned int intnote = note;
 	uint8_t a = intnote % 12;
 	int8_t octave = note / 12.f - 2;
-	uint16_t p1 = amiga_frequencies[a], p2 = amiga_frequencies[a + 1];
+	int32_t p1 = amiga_frequencies[a], p2 = amiga_frequencies[a + 1];
```
Comment (Artefact2) [inline src/play.c:309]: You could multiply here by `AMIGA_FREQ_SCALE` and leave the static array in its original (arguably more readable) form maybe?

Comment (rasky) [reply src/play.c:309]: there are other instances of `amiga_frequencies` lookups, all of them would have to be followed by the multiplication. Is this OK?

Comment (Artefact2) [reply src/play.c:309]: I forgot about that code. Probably best to leave it as-is then :-)

**Comment by Artefact2** [comment]
Merged in 140d42069ab846f827c7c5f53e5930ee50a5e495, thanks for the fix. Modules sounding out of tune is mostly fixed now. (The `amiga.xm` test case is still failing (portas are still broken) against MilkyTracker.)

**Pull request merged**